scipy/spatial/distance.py

```diff
@@ -2653,57 +2889,30 @@ def cdist(XA, XB, metric='euclidean', **kwargs):
     mA = s[0]
     mB = sB[0]
     n = s[1]
-    out = kwargs.pop("out", None)
-    if out is None:
-        dm = np.empty((mA, mB), dtype=np.double)
-    else:
-        if out.shape != (mA, mB):
-            raise ValueError("Output array has incorrect shape.")
-        if not out.flags.c_contiguous:
-            raise ValueError("Output array must be C-contiguous.")
-        if out.dtype != np.double:
-            raise ValueError("Output array must be double type.")
-        dm = out
 
     if callable(metric):
-
         mstr = getattr(metric, '__name__', 'Unknown')
-        metric_name = _METRIC_ALIAS.get(mstr, None)
-
-        XA, XB, typ, kwargs = _validate_cdist_input(XA, XB, mA, mB, n,
-                                                    metric_name, **kwargs)
-
-        for i in range(0, mA):
-            for j in range(0, mB):
-                dm[i, j] = metric(XA[i], XB[j], **kwargs)
-
+        metric_info = _METRIC_ALIAS.get(mstr, None)
+        if metric_info is not None:
+            XA, XB, typ, kwargs = _validate_cdist_input(
```

Comment:
The `metric_info is not None` is already done in `_validate_cdist_input ` no?

Reply:
Removed the check from `_validate_{c,p}dist_input`. All other call sites know for a fact the `metric_info` is valid, which makes more sense to me.